src/api/x509.rs

```diff
@@ -0,0 +1,214 @@
+                error.push_str(None, "'".to_string());
+                error.push_str(None, " not found.".to_string());
+                error.print_data_err();
+            }
+        } else if let (Some(signer_key_path), Some(signer_cert_path)) =
+            (self.inner.signer_key, self.inner.signer_cert)
+        {
+            let signer_key = {
+                let path = Path::new(&signer_key_path);
+                if !path.exists() {
+                    return Err(Error::NonExistingPath {
+                        path: path.to_path_buf(),
+                        source: std::io::Error::new(std::io::ErrorKind::NotFound, "File not found"),
+                    });
+                }
+                KeyPair::from_pem(&fs::read_to_string(path).context(NonExistingPathSnafu { path })?)
+                    .context(CertParamsCreationSnafu)?
+            };
+
+            let signer_cert_pem = {
+                let path = Path::new(&signer_cert_path);
+                if !path.exists() {
+                    return Err(Error::NonExistingPath {
+                        path: path.to_path_buf(),
+                        source: std::io::Error::new(std::io::ErrorKind::NotFound, "File not found"),
+                    });
+                }
+                fs::read_to_string(path).context(NonExistingPathSnafu { path })?
+            };
+
+            let signer_cert = CertificateParams::from_ca_cert_pem(&signer_cert_pem)
+                .context(CertParamsCreationSnafu)?
+                .self_signed(&signer_key)
+                .context(CertificateCreationSnafu)?;
+            params
+                .signed_by(&key_pair, &signer_cert, &signer_key)
+                .context(CertificateCreationSnafu)?
+        } else {
+            params
+                .self_signed(&key_pair)
+                .context(CertificateCreationSnafu)?
+        };
+
+        let cert_pem = cert.pem();
+        let key_pem = key_pair.serialize_pem();
+
+        let out_dir = self
+            .inner
+            .out
+            .unwrap_or_else(|| env::current_dir().unwrap());
+        fs::create_dir_all(&out_dir).unwrap();
+        let cert_filename = format!("{}-certificate.pem", self.inner.common_name);
+        let key_filename = format!("{}-private-key.pem", self.inner.common_name);
+        fs::write(out_dir.join(cert_filename.clone()), &cert_pem).unwrap();
+        fs::write(out_dir.join(key_filename.clone()), &key_pem).unwrap();
+
+        print_json!(&json!({
+            "certificate": out_dir.join(cert_filename),
+            "private_key": out_dir.join(key_filename)
+        }));
+
+        Ok(())
+    }
+}
+
+fn parse_date(date_str: &str) -> Result<OffsetDateTime, Error> {
+    let format = format_description!("[year]-[month]-[day]");
+    time::Date::parse(date_str, &format)
+        .map(|date| date.with_time(time::Time::MIDNIGHT).assume_utc())
+        .map_err(|e| Error::DateParse { source: e })
+}
```

Comment:
You could also create a clap parser, while it's not required, I leave this comment as a possibility c:

Reply:
Ah yeah, appreciate the suggestion. Will move forward with this for now.